### PR TITLE
Add an example-checking script.

### PR DESCRIPTION
**Pull request by jansiegel**
### Context
This PR:
- Makes the examples display the Preview tab by default.
- Adds a docs' example checker script:
    - It finds which `.md` files contain `::: example` containers
    - For those files, it opens their respective URLs (using Puppeteer) and calls a tester function in the context of the web page
    - Currently, the tests checks whether the number of declared Handsontable instances is equal to the actual rendered Handsontable instances
- Implements changes described in https://github.com/handsontable/handsontable/issues/9325.

<sup>[skip changelog]</sup>

### How has this been tested?
Run the script by calling
```
node ./docs/.vuepress/tools/check-examples
```
<sup>(Edited for the changes in react-docs)</sup>

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #9396 

### Affected project(s):
- [ ] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`
- [ ] `@handsontable/vue3`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [x] My change requires a change to the documentation.

### CODE REVIEW

**Comment by jansiegel** [comment]
Notes to the code reviewer (@wszymanski): 
1. It'll be easier to test after https://github.com/handsontable/handsontable/pull/9384 is finished.
2. Testing this requires doing some workarounds:
    - Symlinking the `public/scripts` directory to `dist/docs` to work around the issue with the static content not being copied to the correct place
    - Changing the minimum "frameworked" version to something already existing (like `12.0.0`), because `next` seems to sometimes not generate examples' previews even on the current `develop` (possibly connected with https://github.com/handsontable/handsontable/issues/8787, possibly broken only when building just the `next` version)

**Review by wszymanski — COMMENTED** [review]
> 1. It'll be easier to test after [Implement the docs' snippet framework transformer. #9384](https://github.com/handsontable/handsontable/pull/9384) is finished.

Thus, please make me the reviewer then.

**Comment by jansiegel** [comment]
> Thus, please make me the reviewer then.
You are the reviewer on both of these PRs

File: docs/.vuepress/tools/check-examples/helpers.js [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+ * latest version is available as an url without a version number).
+ *
+ * @param {string} permalink The permalink to be modified.
+ * @param {string} framework The framework to be processed.
+ * @param {string} version The version to be processed.
+ * @returns {string}
+ */
+function extendPermalink(permalink, framework, version) {
+  const latestVersion = getLatestVersion();
+  const frameworkSpecificDocsVersions = getDocsFrameworkedVersions('development');
```
Comment (wszymanski) [inline docs/.vuepress/tools/check-examples/helpers.js:124]: Should it be hardcoded?

**Review by wszymanski — COMMENTED** [review]
Nice idea, code looks good 🎉  Just one question. I'll check how it works after merge.

File: docs/.vuepress/tools/check-examples/helpers.js (hunk 1 above)
Comment (jansiegel) [reply docs/.vuepress/tools/check-examples/helpers.js:124]: It's used just to retrieve the list of all "framework-specific" documentation versions (including `next`), so I think it's fine here.

**Review by wszymanski — COMMENTED** [review]
The script shows errors for every example for me (red dots). I run, for example `node .vuepress/tools/check-examples 12.0` for updated branch. Maybe I'm doing something wrong?

After some thoughts I'm wondering about need of implementation some script which run checks for every existing version of documentation (maybe it's beyond scope of the task). Then running it on the CI would be nice.

**Comment by jansiegel** [comment]
@wszymanski 
> The script shows errors for every example for me (red dots). I run, for example `node .vuepress/tools/check-examples 12.0` for already updated branch. Maybe I'm doing something wrong?

Have you built the docs before running the script? I described it in the JSDoc, but i probably should've done it in the PR description as well, my apologies: https://github.com/handsontable/handsontable/pull/9440/files#diff-1f1d2658b082150ce6290b6e2c6423e5e4fa5e0f255859e2f9e1ea6c76315b24R8

> After some thoughts I'm wondering about need of implementation some script which run checks for every existing version of documentation (maybe it's beyond scope of the task). Then running it on the CI would be nice. What's more maybe npm script for executing test for the latest version would be nice. Without those ideas I'm afraid that script won't be used often.

You're absolutely right, I was thinking of adding such a GHA workflow as well (or extending any of the currently existing ones), but I figured adding one now, when we _know_ some of the examples don't pass the test would make the CI be constantly red. 
My plan was adding it in the PR correcting the examples, as a means of checking if the corrections went well. WDYT?

**Comment by wszymanski** [comment]
> @wszymanski
> 
> > The script shows errors for every example for me (red dots). I run, for example `node .vuepress/tools/check-examples 12.0` for already updated branch. Maybe I'm doing something wrong?
> 
> Have you built the docs before running the script? I described it in the JSDoc, but i probably should've done it in the PR description as well, my apologies: [#9440 (files)](https://github.com/handsontable/handsontable/pull/9440/files#diff-1f1d2658b082150ce6290b6e2c6423e5e4fa5e0f255859e2f9e1ea6c76315b24R8)

Thank you. It is working after running build (with cache clean as we discussed).

> 
> > After some thoughts I'm wondering about need of implementation some script which run checks for every existing version of documentation (maybe it's beyond scope of the task). Then running it on the CI would be nice. What's more maybe npm script for executing test for the latest version would be nice. Without those ideas I'm afraid that script won't be used often.
> 
> You're absolutely right, I was thinking of adding such a GHA workflow as well (or extending any of the currently existing ones), but I figured adding one now, when we _know_ some of the examples don't pass the test would make the CI be constantly red. My plan was adding it in the PR correcting the examples, as a means of checking if the corrections went well. WDYT?

Okey, please do it within another task. You've told me that https://github.com/handsontable/handsontable/issues/9516 may be good for it. What's more please check whether there is a need for bigger timeout.

Screenshot 2022-05-24 at 12 20 33

**Comment by wszymanski** [comment]
Screenshot 2022-06-17 at 12 49 50

It seems that false negative test results are thrown too often.

**Review by wszymanski — COMMENTED** [review]
Screenshot 2022-06-17 at 12 49 50

Everything is okey, beside the fact that false negative test results are thrown too often.

**Review by budnix — APPROVED** [review]
Seems to work 👍 Before merging @jansiegel, please fix all linter issues.

**Review by budnix — CHANGES_REQUESTED** [review]
I've rechecked PR again ... I broke some examples, but after executing the check script, I haven't got any errors. Can you check that?

In Javascript I changed line to
```js
const hot = new Handsonztable(container, {
```

and for React 
```
    <HotzTable
```

**Comment by jansiegel** [comment]
> I've rechecked PR again ... I broke some examples, but after executing the check script, I haven't got any errors. Can you check that?
> 
> In Javascript, I changed a line to
> 
> ```js
> const hot = new Handsonztable(container, {
> ```
> 
> and for React
> 
> ```
> <HotzTable
> ```

@budnix Have you rebuilt the documentation before running the checker script?

**Comment by budnix** [comment]
> @budnix Have you rebuilt the documentation before running the checker script?

Yes 😄

**Comment by jansiegel** [comment]
@budnix, after the changes in https://github.com/handsontable/handsontable/pull/9440/commits/3a083559963acc2eddc116d89f02e56b43178fb6 the issue with typos in `Handsontable`/`HotTable` should be covered as well (in most cases, anyway - when the example container contains a single HOT instance).

**Review by budnix — APPROVED** [review]
👍

**Pull request merged**